db/db_impl.h

```diff
@@ -656,8 +656,11 @@ class DBImpl : public DB {
     autovector<MemTable*> memtables_to_free_;
 
     ~WriteContext() {
-      for (auto& sv : superversions_to_free_) {
-        delete sv;
+      for (auto sv : superversions_to_free_) {
+        if (sv != nullptr) {
+          sv->NotifyOnStallConditionsChanged();
```

Comment:
unfortunately, I think this approach can cause missed notifications. A compaction or flush can finish with `superversions_to_free_` empty because of a user iterator referencing the same `SuperVersion`, and users can hold their iterators indefinitely.